exercises/error-handling/example.py

```diff
@@ -18,5 +18,8 @@ def handle_error_by_returning_tuple(input_data):
 
 def filelike_objects_are_closed_on_exception(filelike_object):
     with filelike_object:
-        filelike_object.close()
-        raise Exception()
+        try:
+            filelike_object.do_something()
+        except Exception:
+            filelike_object.close()
+            raise
```

Comment:
You're actually mixing two error handling methods here - you have a context manager, but then you are explicitly closing an object using `try ... except` blocks to handle the exception behaviour.

Using `try ... finally`: (the `finally` block allows you to close the connection regardless of whether an exception is raised or not, and just doing `except: raise` is redundant)

```python
filelike_object.open()
try:
    filelike_object.do_something()
finally:
    filelike_object.close()
```

Alternatively, a context manager can be used:

```python
with filelike_object:
    filelike_object.do_something()
```

Or a context manager with an assignment:

```python
with filelike_object as infile:
    infile.do_something()
```

Reply:
I had forgotten about `finally`; thanks.